ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+As the ECS is the core part of the architecture, this needs be carefully designed and should be clear under our 3 main vision aspects:
+
+**Modularity** We expose the API from `specs` to the developers, in order to allow them to implement new components, processors and resources (larger singleton objects). The engine will have built-in processors for common operations like `Transform` to ensure spatial hierarchy.
+
+**Parallelism** Processor run in parallel with respect to task- and data-dependencies with low overhead.
+
+**Data-Driven** TODO
+
+**Next steps**
+ * Evaluating different approaches for defining task-dependencies between the processors, including support to run a processor multiple times
+ * Research ways to handle single-threaded processors (e.g nphysics) if possible
+
+### Rendering
+The trend in industry is to achieve realistic rendering, therefore most engines implement physically based rendering.
+Using `gfx-rs` for backend abstraction allows us to provide compatibility with multiple platforms without caring that much about the different graphic APIs
+
+**Modularity** TODO: easily integrate own rendering approaches
+**Parallelism** TODO: parallel command list creation
+**Data-Driven** TODO: efficient rendering -> minimizing draw calls and overhead
+
+**Next steps**
+ * Transfer to physically based rendering (pretty large tasks..)
+ * Evaluate current system in terms of extendability for developers to implement own rendering approaches
+ * Feedback practical experience into improvements for `gfx-rs`
+
+### Input Handling
+**Modularity** The API should be flexible enough to support multiple different input devices including gamepads and maybe VR devices.
+**Data-Driven** TODO: raw input
+
+**Next steps**
+ * Raw input support (winit?)
+ * Gamepad support (winit?)
+ * Input Mapping
+
+### Audio
+TODO: Unfortunately I'm not that deep into the audio components of game engines (state of art?)
+I imagine some kind of OpenAL like API, supporting spatial sound source objects, audio streaming, attenuation, etc.
+Provide low-level access?
+
+** Next Steps**
+
+### Serialization/Reflection
+Type reflection offers additional information about the type, which is available at compile- or run-time. In comparison to C++, to can be achieved easier with Rusts macro system.
+In our engine we want to couple the ECS with serialization/reflection to easily load/save game scenes from various formats. Additionally, it helps us in to build safe network protocols, writing script and GUI bindings.
+
+** Next Steps**
+ * Integrate serde based reflection
+ * Serialize/Deserialize ecs worlds
+
+### Streaming/Asset Handling
+TODO
+
+**Parallelism** TODO: Load assets in parallel
+**Data-Driven** TODO: low loading overhead
+
+** Next Steps**
+ * Improve current approach regarding robustness (error handling) and flexibility with scene formats
+ * Implement loaders for common file formats and archives
```

Comment:
Yes, we can use extisting loaders for formats like `obj`. If they don't exist and we have need for supporting a certain format these libraries then should live in another crate.
The loading code glue for our assetsloader still have to be provided in the engine.